src/xbot/common/logic/Latch.java

```diff
@@ -19,36 +19,32 @@ public Latch(boolean initialValue, EdgeType latchType) {
         this.latchType = latchType;
     }
 
-    public void setValue(boolean value) {
-        boolean different = this.value != value;
+    public void setValue(boolean newValue) {
+        if (this.value == newValue) {
+            return; /* Do nothing if value stays the same */
+        }
 
+        this.value = newValue;
+        EdgeType edgeType = newValue ? EdgeType.RisingEdge : EdgeType.FallingEdge;
+        boolean alertWatchers;
         switch (latchType) {
```

Comment:
Instead of this switch statement, can't we just alert if `latchType.equals(LatchType.Both) || latchType.equals(edgeType)`? Not 100% sure I logicked good as it's currently 4:00 in the morning.

Reply:
Some hard thinking I figured out how to shorten most things.